### ISSUE TITLE
`unused_import` seems to work only for public imports

### ISSUE THREAD

**Issue by rgoldberg** [realm/SwiftLint#6620]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`unused_import` seems to work only for public imports.

If correct, it should either be fixed to work for all imports, or be documented as only working for public imports.

```swift
internal import Foundation // does not trigger a violation
```

zsh:

```shell
swiftlint analyze --compiler-log-path =(xcodebuild -scheme myscheme -destination "platform=macOS,arch=$(arch),variant=macos" 2>&1)
```

### Environment

* SwiftLint version: 0.63.2
* Xcode version:  Xcode 26.4.1 / Build version 17E202
* Installation method used: Homebrew
* Configuration file:

```yml
analyzer_rules:
- unused_import
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
